query/cursor.go

```diff
@@ -233,6 +233,12 @@ func newScannerCursor(s IteratorScanner, fields []*influxql.Field, opt IteratorO
 
 func (s *scannerCursor) scan(m map[string]interface{}) (int64, string, Tags) {
 	ts, name, tags := s.scanner.Peek()
+	// if a new series, clear the map of previous values
+	if name != s.series.Name || tags.ID() != s.series.Tags.ID() {
+		for k := range m {
```

Comment:
Can you encapsulate this logic with a `clear(m)` method on the `scannerCursorBase`?